src/dhcp/dhcp_config_utils.c

```diff
@@ -11,51 +11,37 @@ bool get_config_dhcpinfo(const char *info, config_dhcpinfo_t *el) {
   UT_array *info_arr;
   utarray_new(info_arr, &ut_str_icd);
 
-  if (split_string_array(info, ',', info_arr) < 0) {
+  ssize_t substrings = split_string_array(info, ',', info_arr);
+  if (substrings < 5) {
```

Comment:
Ideally we should create a proper parser for these lines.

Reply:
I've changed the function to use `sscanf()` in https://github.com/nqminds/edgesec/pull/549/commits/0f069b43d7220b790e3d029d886c2d887b10a1aa, but it requires some of the C pre-processor dark arts, since if `OS_INET_ADDRSTRLEN` is `22`, you can't use `%22s`, it has to be `%21s` due to the NUL-terminator.

```c
  int substrings =
      sscanf(info,
             "%d," SCANF_IPv4_STRING "," SCANF_IPv4_STRING "," SCANF_IPv4_STRING
             "," SCANF_DHCP_LEASE_TIME_STRING,
             &scanf_results.vlanid, scanf_results.ip_addr_low,
             scanf_results.ip_addr_upp, scanf_results.subnet_mask,
             scanf_results.lease_time);
```

Use `sscanf` is much nicer, since it will fail on invalid IPs and strings that are too long!